### PR TITLE
[EDS-556] Return student search results when requester is authenticated

### PR DESCRIPTION
**Pull request by tomthorogood**
This is a larger PR that closes out EDS-556. It is broken into 3 commits. I recommend reading each individual commit message for more details of what this changes. 

The broad strokes:

- [Updates the backend models to be cognizant of the new "populations" layer](8d8230238b34f7ef361734859e4d74ffe5ef76fc)
- [Updates the frontend to use the new backend models](72d0a81b49699f4658b0e4ae0597c1cac41b4151)
- [Updates (and adds) tests for the new behavior](e81700d)

### CODE REVIEW

File: husky_directory/templates/summary_results_table.html
Comment (goulter) [inline husky_directory/templates/summary_results_table.html:48]: Please correction the indentation of the for loops and if statement.

**Review by goulter — CHANGES_REQUESTED** [review]

File: husky_directory/templates/summary_results_table.html
Comment (tomthorogood) [reply husky_directory/templates/summary_results_table.html:48]: I did it this way on purpose as an experiment. happy to put it back to "normal"

Comment (goulter) [reply husky_directory/templates/summary_results_table.html:48]: Yes, please do.

File: husky_directory/templates/result_count.html
Comment (tomthorogood) [inline husky_directory/templates/result_count.html:15]: I think I need to get rid of this `<br>`:

Screen Shot 2021-03-29 at 3 57 43 PM

should look like:

Screen Shot 2021-03-29 at 3 58 54 PM

**Comment by tomthorogood** [comment]
Updated, please see [latest commit message](https://github.com/UWIT-IAM/uw-husky-directory/pull/47/commits/53c855df5c350bb04c6d151fe1b9a7d5b1738d87)

**Comment by goulter** [comment]
Cool, I'll start back on this in the A.M with a fresh brain.

**Review by goulter — CHANGES_REQUESTED** [review]
Search result page with 0 results always puts the search type to "Name" regardless of the original search type.
There are 2 behaviors that happen under this scenario:
Search "Goulter" and set type to anything expect name
- result is No matches for is ""  (Note, the no matches part is correct)
- search type is set to name

Search "Goulter" and set type to name
- result No matches for is "Goulter" . (Note, the no matches part is correct)

**Comment by tomthorogood** [comment]
Putting this PR on hold in lieu of #48.

I will update this one and re-request review once #48 is approved.

**Comment by tomthorogood** [comment]
ok, this PR has been updated to include the changes from #48, and now the PR image linked here has the same behavior.

**Review by goulter — COMMENTED** [review]
Running locally, the sign in link reappears after I click it and do a search. In a prior version/commit, the link remained.
I don't really think that's a blocker for this PR since this is using a mock login.
Otherwise, looks pretty good in terms of the student enabled searches.
@tomthorogood just let me know and I'll approve the PR :)

**Comment by tomthorogood** [comment]
hmm, there is a test for the sign in link persistence. let me take a look, i don't want to commit a regression to something that i've already fixed 3 times.

**Comment by tomthorogood** [comment]
Ok, I can repro and identified why my the test didn't catch it. Thanks again!

**Comment by goulter** [comment]
You're welcome and thank you for rolling with it :)

**Comment by tomthorogood** [comment]
Ok, fixed again! And I added a test to catch henceforth.

**Review by goulter — CHANGES_REQUESTED** [review]
I should have caught this earlier, but its a simple one. The form's input field has [code](https://github.com/UWIT-IAM/uw-husky-directory/blob/main/husky_directory/templates/index.html#L65) in there to be auto-focused if the field is empty, like on a new search. The console and page behavior say we broke it. Should be an easy fix, in that line, change `dirform.name` to `dirform.query` and I think we'll be golden...maybe :)

**Comment by tomthorogood** [comment]
Fixed, and as mentioned offline (but noting here for visibility), it'd be great if you could create a jira with these manual test cases that you'd like to automate.

**Review by goulter — CHANGES_REQUESTED** [review]
I found one more place affecting search results. In the summary results, the `More` button generates a `name` search via the old field `name` instead of `query`.
All you should need is change `name = "name"` to `query = "name"` [here](https://github.com/UWIT-IAM/uw-husky-directory/blob/main/husky_directory/templates/summary_results.html#L38)

**Comment by goulter** [comment]
> create a jira with these manual test cases that you'd like to automate.
Great idea! Hopefully this will reduce the back and forth and check for common bugs.
https://jira.cac.washington.edu/browse/EDS-565

**Comment by tomthorogood** [comment]
Are there any other test cases you need to run before I make another change? I would prefer not to keep finding them one at a time.

**Comment by goulter** [comment]
There a likely several smaller test cases that I will document as I think of them.
I understand your concern.
How about we get this last one fixed and if the cases I've been running all pass, we merge. 
If anything outside of those discovers a bug, I'll file a JIRA bug for it and add to the test cases needed.

**Comment by tomthorogood** [comment]
In that case I would prefer to just defer other fixes. This isn't getting us anywhere, and i don't think the fact that we have incomplete test cases should continue blocking this PR. 

I fulfilled the contract of expanding tests for additional functionality, all of which passed, and expanding tests for new test cases that emerged. I think at this point, a better option is to merge this PR, and then work on automating the test cases as a separate jira, which you've already created. That will be responsible for fixing every little thing. The fact that there were test cases of which I was never made aware shoudln't be a punishment to the effort to return student search results.

**Review by goulter — APPROVED** [review]
We took this discussion offline. Tom and I are on the same page now.

**Pull request merged**
